--- a/arch/arm64/Makefile
+++ b/arch/arm64/Makefile
@@ -158,8 +158,12 @@ endif
 all:	$(notdir $(KBUILD_IMAGE))
 
 
+DTBS := rk35*-nanopi*.dtb
 Image vmlinuz.efi: vmlinux
 	$(Q)$(MAKE) $(build)=$(boot) $(boot)/$@
+	$(Q)scripts/mkkrnlimg $(objtree)/arch/arm64/boot/Image $(objtree)/kernel.img >/dev/null
+	@echo '  Image:  kernel.img is ready'
+	$(Q)$(srctree)/scripts/mkimg --dtb $(DTBS)
 
 Image.%: Image
 	$(Q)$(MAKE) $(build)=$(boot) $(boot)/$@
--- a/scripts/Makefile
+++ b/scripts/Makefile
@@ -10,6 +10,7 @@ hostprogs-always-$(CONFIG_BUILDTIME_TABL
 hostprogs-always-$(CONFIG_ASN1)				+= asn1_compiler
 hostprogs-always-$(CONFIG_MODULE_SIG_FORMAT)		+= sign-file
 hostprogs-always-$(CONFIG_SYSTEM_EXTRA_CERTIFICATE)	+= insert-sys-cert
+hostprogs-always-$(CONFIG_ARM64)			+= resource_tool mkkrnlimg
 hostprogs-always-$(CONFIG_RUST)				+= generate_rust_target
 
 generate_rust_target-rust := y
